cli_tools/common/utils/daisyutils/daisy_worker.go

```diff
@@ -74,35 +78,93 @@ func createResourceLabelerIfMissing(env EnvironmentSettings, hooks []interface{}
 }
 
 type defaultDaisyWorker struct {
-	wf     *daisy.Workflow
-	logger logging.Logger
-	env    EnvironmentSettings
-	hooks  []interface{}
+	workflowProvider WorkflowProvider
+	finished         *daisy.Workflow
+	logger           logging.Logger
+	env              EnvironmentSettings
+	hooks            []interface{}
+
+	cancel      chan string
+	cancelGuard sync.Once
 }
 
 // Run runs the daisy workflow with the supplied vars.
-func (w *defaultDaisyWorker) Run(vars map[string]string) error {
-	if err := (&ApplyAndValidateVars{w.env, vars}).PreRunHook(w.wf); err != nil {
-		return err
+func (w *defaultDaisyWorker) Run(vars map[string]string) (err error) {
+	var wf *daisy.Workflow
+	for attempts := 0; attempts <= 1; attempts++ {
```

Comment:
Should attempt count limit be configurable? I assume one retry attempt is enough for use cases we currently handle, but, since you're anyway adding the concept of retry, it might make sense to make it configurable and default to 2 attempts.

Reply:
For the current requirements, I don't think multiple retries would be helpful. That said, I was thinking the refactoring would be easy if someone needed to pull that out later.